src/megaclient.cpp

```diff
@@ -12565,13 +12565,19 @@ bool MegaClient::syncup(LocalNode* l, dstime* nds)
 
         rit = nchildren.find(&localname);
 
-        string localpath;
-        unique_ptr<FileAccess> fa(fsaccess->newfileaccess(false));
-
-        ll->getlocalpath(&localpath);
-        fa->fopen(&localpath);
-        bool isSymLink = fa->mIsSymLink;
+        bool isSymLink = false;
+#ifndef WIN32
+        if (PosixFileAccess::mFoundASymlink)
+        {
+            string localpath;
+            unique_ptr<FileAccess> fa(fsaccess->newfileaccess(false));
 
+            ll->getlocalpath(&localpath);
+            fa->fopen(&localpath);
+            isSymLink = fa->mIsSymLink;
+            LOG_debug << "checking if localpath is symlink: " << localpath;
```

Comment:
Although it is useful to know if we are checking that, outputting the message here for every check may overwhelm the log, since syncup() will call this for every single file/folder in the sync as it recurses through the tree - for very large accounts syncing everything, that will make it hard to read the logs, and cause more CPU and I/O.

Reply:
Agree. We can have the log message only when we first set mFoundASymlink to true instead